jenkins/custom-ci/generic/cortx-cc.groovy

```diff
@@ -0,0 +1,114 @@
+        string(name: 'CORTX_CC_URL', defaultValue: 'https://github.com/Seagate/cortx-cc', description: 'Repository URL for cortx-cc build')
+        string(name: 'CORTX_CC_BRANCH', defaultValue: 'main', description: 'Branch for cortx-cc build')
+        string(name: 'CORTX_UTILS_URL', defaultValue: 'https://github.com/Seagate/cortx-utils', description: 'CORTX Utils Repository URL', trim: true)
+        string(name: 'CORTX_UTILS_BRANCH', defaultValue: 'main', description: 'Branch or GitHash for CORTX Utils', trim: true)
+        string(name: 'THIRD_PARTY_PYTHON_VERSION', defaultValue: 'custom', description: 'Third Party Python Version to use', trim: true)
+        string(name: 'CUSTOM_CI_BUILD_ID', defaultValue: '0', description: 'Custom CI Build Number')
+        choice(
+            name: 'BUILD_LATEST_CORTX_CC',
+                choices: ['yes', 'no'],
+                description: 'Build cortx-cc from latest code or use last-successful build.'
+            )
+        // Add os_version parameter in jenkins configuration    
+    }
+    
+
+       environment {
+        release_dir = "/mnt/bigstorage/releases/cortx"
+        component = "cortx-cc"
+        release_tag = "custom-build-$CUSTOM_CI_BUILD_ID"
+        build_upload_dir = "$release_dir/github/integration-custom-ci/$os_version/$release_tag/cortx_iso"
+        python_deps = "${THIRD_PARTY_PYTHON_VERSION == 'cortx-2.0' ? "python-packages-2.0.0-latest" : THIRD_PARTY_PYTHON_VERSION == 'custom' ?  "python-packages-2.0.0-custom" : "python-packages-2.0.0-stable"}"
+    }
+    
+    
+    
+    options {
+        timeout(time: 30, unit: 'MINUTES')
+        timestamps()
+        buildDiscarder(logRotator(daysToKeepStr: '5', numToKeepStr: '10'))
+    }
+
+    stages {
+    
+        stage('Checkout cortx-cc') {
+            when { expression { params.BUILD_LATEST_CORTX_CC == 'yes' } }
+            steps {
+                script { build_stage = env.STAGE_NAME }
+                dir ('cortx-cc') {
+                    checkout([$class: 'GitSCM', branches: [[name: "${CORTX_CC_BRANCH}"]], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'CloneOption', depth: 0, noTags: false, reference: '', shallow: false, timeout: 15], [$class: 'SubmoduleOption', disableSubmodules: false, parentCredentials: true, recursiveSubmodules: true, reference: '', trackingSubmodules: false, timeout: 15]], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'cortx-admin-github', url: "${CORTX_CC_URL}"]]])
+                }
+            }
+        }
+    
+        stage('Install Dependencies') {
+            when { expression { params.BUILD_LATEST_CORTX_CC == 'yes' } }
+            steps {
+                script { build_stage = env.STAGE_NAME }
+
+                sh label: 'Install cortx-cc pre-requisites', script: '''
+                    yum -y install python3 python3-devel facter yum-utils
+                    yum-config-manager --add-repo https://rpm.releases.hashicorp.com/RHEL/hashicorp.repo
```

Comment:
Use local repo for `consul` installation - Refer - http://cortx-storage.colo.seagate.com/releases/cortx/third-party-deps/rockylinux/rockylinux-8.4-2.0.0-latest/hare/

Reply:
Done